spec/models/conversion_host_spec.rb

```diff
@@ -239,4 +239,15 @@
       expect(conversion_host.errors[:resource].first).to eql("can't be blank")
     end
   end
+
+  context "name validation" do
+    let(:ems) { FactoryBot.create(:ems_redhat, :zone => FactoryBot.create(:zone), :api_version => '4.2.4') }
+    let(:redhat_host) { FactoryBot.create(:host_redhat, :name => 'foo', :ext_management_system => ems) }
+
+    it "defaults to the associated resource name if no name is explicitly provided" do
```

Comment:
Don't you also test with an OpenStack VM as resource ?

Reply:
For purposes of a validation spec it's not really necessary. We mostly just want to make sure the name is set, regardless of the associated resource type.